conbench/entities/run.py

```diff
@@ -497,6 +515,49 @@ def change_pr_number_empty_string_to_none(self, data, **kwargs):
         },
     )
 
+    @marshmallow.validates_schema
+    def validate_props(self, data, **kwargs):
+        url = data["repository"]
+        if not url.startswith("https://github.com"):
+            raise marshmallow.ValidationError(
+                f"'repository' must be a URL, starting with 'https://github.com', got `{url}`"
```

Comment:
There's some legacy code where this will break: If users opt-in to github info autodetection (which we discourage), it will pass their git remote (which may start with `git@`) here: https://github.com/conbench/conbench/blob/e1cab9b8ebffaba1c85e454b1b13de7f14bfcf02/conbench/machine_info.py#L127 

plus in benchadapt. If we're not going to accept that, we should probably move the transform-to-URL code to the client-side, or remove the autodetection entirely.

Reply:
Though I'm not sure there's anywhere that's using autodetection today, it does seem like a useful feature to keep. I'd rather move the URL-transformation to benchadapt.

Before merging this PR, we also need to think about (and, IMO, implement!) how to supply "no commit info" through benchadapt. Right now if you do `github=None` it opts you in to autodetection. Maybe that needs to be rethought. We can't allow benchadapt to get to a state where it can't supply "no commit info" to Conbench.

Reply:
Thanks for feedback!

> We can't allow benchadapt to get to a state where it can't supply "no commit info" to Conbench

I thought maybe benchadapt is guaranteed to send plausible commit info or it does error out.

OK, looks like we're not so strict: https://github.com/conbench/conbench/blob/e1cab9b8ebffaba1c85e454b1b13de7f14bfcf02/benchadapt/python/benchadapt/result.py#L191

OK, so _there_ we need to hook in and use the 'new' way of communicating "not commit info". OK, I am on that.

Reply:
> There's some legacy code where this will break: If users opt-in to github info autodetection (which we discourage), it will pass their git remote (which may start with git@) here:

Thanks for letting me know.

To short-cut this conversation/topic I am adding a commit here that for now accepts this URL format and transparently re-writes this into the `https://github.com/...` format.